bedrock/contentful/api.py

```diff
@@ -447,51 +452,120 @@ def page(self):
+            {
+                "title": title,
+                "blurb": blurb,
+                "slug": slug,
+                "locale": locale,
+                "theme": "firefox" if "firefox" in folder else "mozilla",
+                # eg www.mozilla.org-firefox-accounts or www.mozilla.org-firefox-sync
+                "utm_source": f"www.mozilla.org-{campaign}",
+                "utm_campaign": campaign,  # eg firefox-sync
+            }
+        )
 
-        if "preview_image" in fields:
-            # TODO request proper size image
-            preview_image_url = fields["preview_image"].fields().get("file").get("url")
-            data["image"] = "https:" + preview_image_url
+        _preview_image = self._get_preview_image_from_fields(fields)
+        if _preview_image:
+            data["image"] = _preview_image
+
+        if seo_fields:
+            _preview_image = self._get_preview_image_from_fields(seo_fields)
+            if _preview_image:
+                seo_fields["image"] = _preview_image
+                del seo_fields["preview_image"]
+            data.update({"seo": seo_fields})
+
+        # TODO: Check with plans for Contentful use - we may
+        # be able to relax this check and use it for page types
+        # once we're in all-Compose mode
+        if page_type == CONTENT_TYPE_PAGE_RESOURCE_CENTER:
+            if "category" in fields:
+                data["category"] = fields["category"]
+            if "tags" in fields:
+                data["tags"] = fields["tags"]
+            if "product" in fields:
+                # NB: this is a re-mapping with an eye on flexibility - pages may not always have
+                # a 'product' key, but they might have something regarding overall classification
+                data["classification"] = fields["product"]
 
         return data
 
     def get_entry_by_id(self, entry_id):
         return self.client.entry(entry_id, {"locale": self.locale})
 
     def get_content(self):
-        # check if it is a page or a connector
+        # Check if it is a page or a connector, or a Compose page type
+
         entry_type = self.page.content_type.id
-        if entry_type.startswith("page"):
+        seo_obj = None
+        if entry_type == COMPOSE_MAIN_PAGE_TYPE:
+            # Contentful Compose page, linking to content and SEO models
+            entry_obj = self.page.content  # The page with the actual content
+            seo_obj = self.page.seo  # The SEO model
+            # Note that the slug lives on self.page, not the seo_obj.
+        elif entry_type.startswith("page"):
             entry_obj = self.page
         elif entry_type == "connectHomepage":
+            # Legacy - TODO: remove me once we're no longer using Connect: Homepage
             entry_obj = self.page.fields()["entry"]
         else:
             raise ValueError(f"{entry_type} is not a recognized page type")
 
-        self.request.page_info = self.get_info_data(entry_obj)
+        if not entry_obj:
```

Comment:
It was added because there's a Compose page with only an SEO block and no content page set, which did indeed cause a blow-up. Here's my local output in non-quiet mode

```Running forced update from Contentful data
Pulling from Contentful
Updating existing ContentfulEntry for connectHomepage:58YIvwDmzSDjtvpSqstDcL
Updating existing ContentfulEntry for connectHomepage:4k3CxqZGjxXOjR1I0dhyto
Updating existing ContentfulEntry for pagePageResourceCenter:5wIcW1esdWIK6YYtfvhLL1
Updating existing ContentfulEntry for pagePageResourceCenter:IORNDWEoanCkQ0TwKhlfB
Updating existing ContentfulEntry for pagePageResourceCenter:4Gobq3NdbOCaDyLapsDIJR
Updating existing ContentfulEntry for pagePageResourceCenter:3gikThlF98f2hekURp4N51
Updating existing ContentfulEntry for pagePageResourceCenter:1dQAhsz5Xnr6IrGwvhnFZp
Updating existing ContentfulEntry for pagePageResourceCenter:1JxYJUTGphwTNR9iofafkG
Updating existing ContentfulEntry for pageHome:1XWEqmSvQQF6eCqNOTAGUL
Updating existing ContentfulEntry for pagePageResourceCenter:2LEDujIfxanZKMH3Us0QXO
Problem with page:3bKxRzCv6kHjCAswfZKG3a -> <class 'Exception'>: No 'Entry' detected for page
Updating existing ContentfulEntry for pageHome:60YAs15SYSKKnjtZ4x3SO7
Updating existing ContentfulEntry for pageHome:1HercJWjhIEm4SxeyWLTYF
Done. Added: 0. Updated: 12. Errors: 1```

Reply:
OH! I see. So line 552 could result in `entry_obj` still being `None`. Got it. Thanks.